crates/typst-library/src/text/raw.rs

```diff
@@ -247,7 +266,14 @@ impl Synthesize for RawElem {
 impl Show for RawElem {
     #[tracing::instrument(name = "RawElem::show", skip_all)]
     fn show(&self, _: &mut Vt, styles: StyleChain) -> SourceResult<Content> {
-        let text = self.text();
+        let tab_size = RawElem::tab_size_in(styles) as usize;
+        let text = if tab_size > 0 && self.text().contains('\t') {
+            let replacement =
+                unsafe { String::from_utf8_unchecked(vec![b' '; tab_size]) };
+            self.text().replace('\t', &replacement).into()
+        } else {
+            self.text()
+        };
```

Comment:
```suggestion
        let mut text = self.text();
        if text.contains('\t') {
            let tab_size = RawElem::tab_size_in(styles) as usize;
            let replacement = " ".repeat(tab_size);
            text = text.replace('\t', &replacement).into();
        }
```
1. Better to retrieve `text()` just once.
2. Better to retrieve tab size only if necessary. Getting rid of the tabs if tab-size == 0 seems better than ignoring them to me.
3. No need to use unsafe here. Even if it would be a bit slower (I doubt that it's considerably slower).